### PR TITLE
Updated Simple Gain model to new framework with channel based processing

### PR DESCRIPTION
**Pull request by fe-wickham**
Notably the framework has been unable to generate working channel basing processing models for some time. In particular, the valid and channel signals were being sampled at the data rate but as the values are only pulsed the pulses weren't in sync with when they were being sampled they were always missed. With this model I found making the Avalon_Data_Processing triggered based off the valid signal fixes these problems.

### CODE REVIEW

File: models/simple_gain/modelparameters.m [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+%% Exponential moving average setup
+
+% 10 ms window
+mp.windowSize = 10e-3 * mp.Fs;
+
+% define the exponential moving average weight to be roughly equivalent to
+% a simple moving average of length mp.windowSize
+% https://en.wikipedia.org/wiki/Moving_average#Relationship_between_SMA_and_EMA
+mp.exponentialMovingAverageWeight = fi(2/(mp.windowSize + 1), 0, 32, 31);
+
```
Comment (tvannoy) [inline models/simple_gain/modelparameters.m]: These lines do not apply to the simple gain model

**Review by tvannoy — CHANGES_REQUESTED** [review]

File: models/simple_gain/model.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            "registers": [
+                {
+                    "dataType": {
+                        "fractionLength": 24,
+                        "signed": false,
+                        "type": "fix32",
+                        "wordLength": 32
+                    },
+                    "defaultValue": 1.5,
+                    "name": "Left_Gain",
```
Comment (tvannoy) [inline models/simple_gain/model.json]: Since this gain now applies to both channels, let's change the name here and in the model.

**Review by tvannoy — CHANGES_REQUESTED** [review]
Model framework looks good. Just a couple really minor things to change and then this is good to go.

File: models/simple_gain/model.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "devices": [
+        {
+            "name": "simple_gain",
+            "registers": [
+                {
+                    "dataType": {
+                        "fractionLength": 24,
+                        "signed": false,
+                        "type": "fix32",
+                        "wordLength": 32
```
Comment (tvannoy) [inline models/simple_gain/model.json]: We could consider making this data type smaller, as we don't really need all of these bits for a gain; I doubt we would hear any differences at 2^-24...

Comment (fe-wickham) [reply models/simple_gain/model.json]: Thats fair.

**Pull request merged**